packages/core-plugin-api/src/plugin-options/usePluginOptions.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createVersionedContext,
+  createVersionedValueMap,
+  useVersionedContext,
+} from '@backstage/version-bridge';
+import { BackstagePlugin, PluginOptions } from '../plugin';
+import React, { ReactNode } from 'react';
+
+const contextKey: string = 'plugin-options-context';
+
+/**
+ * Properties for the AsyncEntityProvider component.
+ *
+ * @public
+ */
+export interface PluginOptionsProviderProps {
+  children: ReactNode;
+  plugin?: BackstagePlugin;
+}
+
+export const PluginProvider = ({
+  children,
+  plugin,
+}: PluginOptionsProviderProps): JSX.Element => {
+  const providerPlugin = plugin as unknown as {
+    getPluginOptions(): PluginOptions;
+  };
+  const value = { pluginOptions: providerPlugin.getPluginOptions() };
```

Comment:
I'm thinking it's more forwards compatible to send the plugin through the context and extract the options in the consumer hook instead, what do you think?